### PR TITLE
Workflow Support `op.# ApplyComponent` and `op.#ApplyRemaining`

### PR DESCRIPTION
**Pull request by leejanee**
1. Optimize the ability of workflow as a base platform
2. Support `op.# ApplyComponent` and `op.#ApplyRemaining`  with [cue configuration](https://github.com/leejanee/kubevela/blob/workflow-apply/pkg/stdlib/op.go).

```
#ApplyComponent: op.#Steps & {
   component: string
   _componentName:  component

   load:  op.#Load & {
      component: _componentName
   }

   workload: op.#Apply & {
      // Easy to read status
      status: value.status
      value: load.value.workload
      ...
   }
    
   #up: [ for o in load.value.auxiliaries {
          op.#Apply & { value: o }
   }]
}
```

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: docs/examples/workflow/definition.yaml [hunk 1]
```diff
@@ -1,39 +1,70 @@
 apiVersion: core.oam.dev/v1beta1
 kind: WorkflowStepDefinition
 metadata:
-  name: apply
+  name: apply-withIP
```
Comment (hongchaodeng) [inline docs/examples/workflow/definition.yaml:4]: apply-with-ip

AFAIK, name doesn't separate lower/upper cases.

File: docs/examples/workflow/apply-remaining/definition.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+metadata:
+  name: apply-component
+spec:
+  schematic:
+    cue:
+      template: |
+        import ("vela/op")
+        parameter: {
+           component: string
+           prefixIP?: string
```
Comment (hongchaodeng) [inline docs/examples/workflow/apply-remaining/definition.yaml:12]: What's prefixIP for? Can you add some comment for it?

File: docs/examples/workflow/apply-remaining/definition.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  name: apply-remaining
+spec:
+  schematic:
+    cue:
+      template: |
+        import ("vela/op")
+        parameter: {
+            exceptions?: [componentName=string]: {
+            // skipApplyWorkload indicates whether to skip apply the workload resource
+            skipApplyWorkload: *true | bool
```
Comment (hongchaodeng) [inline docs/examples/workflow/apply-remaining/definition.yaml:45]: pay attention to indentation.

**Review by hongchaodeng — APPROVED** [review]
The code lgtm.
Have some questions.

**Comment by wonderflow** [comment]
@hongchaodeng yes, we should refine the examples.  Will merge this PR for now. Feel free to send following PRs.

**Pull request merged**